app/models/message.rb

```diff
@@ -261,6 +276,80 @@ def should_index?
 
   private
 
+  def flatten_custom_attributes(attrs)
+    return [] if attrs.blank?
+
+    attrs.map do |key, value|
+      {
+        key: key.to_s,
+        value: value.to_s,
+
+        value_type: value.class.name.downcase
+      }
+    end
+  end
+
+  def search_attachments_data
+    attachments.filter_map do |a|
+      { transcribed_text: a.meta&.dig('transcribed_text') }
```

Comment:
`transcribed_text` would be `nil` for most of the messages. Should we create the field only if `transcribed_text.present?`